playbooks/robusta_playbooks/job_restart_on_oomkilled.py

```diff
@@ -0,0 +1,196 @@
+            backoffLimit=job_event.spec.backoffLimit,
+            activeDeadlineSeconds=job_event.spec.activeDeadlineSeconds,
+            ttlSecondsAfterFinished=job_event.spec.ttlSecondsAfterFinished,
+            template=PodTemplateSpec(
+                spec=PodSpec(
+                    containers=container_list,
+                    restartPolicy=job_event.spec.template.spec.restartPolicy,
+                    nodeName=job_event.spec.template.spec.nodeName,
+                    activeDeadlineSeconds=job_event.spec.template.spec.activeDeadlineSeconds,
+                    nodeSelector=job_event.spec.template.spec.nodeSelector,
+                    affinity=job_event.spec.template.spec.affinity,
+                    initContainers=job_event.spec.template.spec.initContainers,
+                    serviceAccount=job_event.spec.template.spec.serviceAccount,
+                    securityContext=job_event.spec.template.spec.securityContext,
+                    volumes=job_event.spec.template.spec.volumes,
+                ),
+            ),
+        ),
+    )
+    job_event.delete()
+    job_spec.create()
+    return job_spec
+
+
+# function to get Containers attributes
+def get_container_list(containers_spec, increase_to):
+    containers_list = []
+    for container in containers_spec:
+        containers_list.append(
+            Container(
+                name=container.name,
+                image=container.image,
+                livenessProbe=container.livenessProbe,
+                securityContext=container.securityContext,
+                volumeMounts=container.volumeMounts,
+                args=container.args,
+                command=container.command,
+                ports=container.ports,
+                lifecycle=container.lifecycle,
+                readinessProbe=container.readinessProbe,
+                workingDir=container.workingDir,
+                env=container.env,
+                startupProbe=container.startupProbe,
+                envFrom=container.envFrom,
+                imagePullPolicy=container.imagePullPolicy,
+                resources=increase_resource(container.resources, increase_to)
+                if (container.resources.limits and container.resources.requests)
+                else None,
+            )
+        )
+    return containers_list
+
+
+# Function to increase resources
+def increase_resource(resource, increase_to):
+    limits = resource.limits["memory"]
+    reqests = resource.requests["memory"]
+
+    split_lim, lim_unit = split_num_and_str(limits)
+    split_req, req_unit = split_num_and_str(reqests)
+
+    split_req = float(split_req) + float(increase_to)
+
+    if split_req > float(split_lim):
+        split_lim = split_req
+
+    return ResourceRequirements(
+        limits={"memory": (str(split_lim) + lim_unit)},
+        requests={"memory": (str(split_req) + req_unit)},
+    )
+
+
+# Function to get job's Pod
+def get_job_pod(namespace, job):
```

Comment:
Actually, the reason why I am searching for the oomkilled container in a pod is to increase the memory of the right container,
but in the case of get_job_pod, I am finding the pod related to this job (there can be other pods as well) and it also returns the right pod because the trigger calls only when the job's pod failed.

"get_job_pod" has been changed to “get_job_latest_pod” from job_utils.